subtests/docker_cli/info/info.py

```diff
@@ -40,38 +41,81 @@ def run_once(self):
 
     @staticmethod
     def _build_table(cli_output):
-        out = cli_output.split('\n')
-        out = [x.split(':', 1) for x in out if x]
-        keys = [x[0].strip() for x in out]
-        vals = []
-        for x in out:
-            try:
-                vals.append(x[1].strip())
-            except IndexError:
-                vals.append(None)
-        return dict(zip(keys, vals))
+        """
+        'docker info' returns a human-readable list of key: value pairs.
+        Some are indented by a space, indicating that these key/values
+        are subelements of a previous element. To wit:
+
+             Images: 3
+             Server Version: 1.12.6
+             Storage Driver: devicemapper
+              Pool Name: vg--docker-docker--pool
+              Pool Blocksize: 524.3 kB
+              ...
+             Logging Driver: journald
+
+        In this case 'Images', 'Server Version', and 'Logging Driver'
+        are simple tuples but 'Storage Driver' has both a value and
+        a set of further key/value pairs: Pool Name, Pool Blocksize, ...
+
+        We parse that and return a dict with the expected key/value
+        mapping *and* an extra: for all elements with subelements,
+        'element...' (element name plus three dots) is a dict containing
+        the subelements. E.g. x['Storage Driver...']['Pool Name'] = 'vg--etc'
+        """
+        out = {}
+        current_key = None
+        for line in cli_output.split('\n'):
+            # Almost every line will be Foo: Bar, but 'Insecure Registries:'
+            # is followed by a simple list of IPv4 netmasks
+            if ': ' in line or line.endswith(':'):
+                (key, value) = [e.strip() for e in line.split(':', 1)]
+            else:
+                (key, value) = (line.strip(), '')
+            if line.startswith(' '):
+                if not current_key:
+                    raise IndexError("sdfsdf")
+                if current_key not in out:
+                    out[current_key] = {}
+                out[current_key][key] = value
+            else:
+                out[key] = value
+                current_key = key + '...'
```

Comment:
Dealing with this:

    Storage Driver: overlay2
        Foo: bar
        Bar: baz

You can't just set `out['Storage Driver']` to a dict, because you lose the `= overlay2` value. I could set it to a dict with something like `['__value'] = overlay2` but that feels clumsy. I could create a new dict-based class that treats string compares magically, but I'm not that insane. This seemed the least evil approach.